J/jlqml/build_tarballs.jl

```diff
@@ -2,14 +2,17 @@
 # `julia build_tarballs.jl --help` to see a usage message.
 using BinaryBuilder, Pkg
 
-julia_version = v"1.6.0"
+uuid = Base.UUID("a83860b7-747b-57cf-bf1f-3e79990d037f")
+delete!(Pkg.Types.get_last_stdlibs(v"1.6.3"), uuid)
```

Comment:
Doesn't this actually break everything when we try to target Julia v1.6?  I don't see `LibOSXUnwind_jll` in https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=16459&view=logs&jobId=3e3b6bc7-f97f-5652-f6ca-c52d13eda498&j=a151bd49-d959-5dd9-e79c-4da26ad24f9f&t=fa03432c-08c9-5f99-931b-dc1b3c71632b, which looks wrong.

Reply:
As far as I understand, `LibOSXUnwind_jll` is a dependency of `libjulia_jll` and removing it from the `stdlibs` makes Pkg choose the one in the registry. It doesn't appear to cause harm, at least on Linux I get a usable JLL with binaries for Julia 1.6, 1.7 and 1.8: https://github.com/barche/jlqml_jll.jl

Reply:
I still don't see `LibOSXUnwind_jll` in https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=16556&view=logs&j=11cc9737-c3b2-5b2e-49c1-3ecd4dd3473d&t=6864626c-b206-5daa-5d51-831805962223.  @fingolfin maybe it wasn't necessary after all?  If so, I think the best thing to do would be to have a `libjulia_jll` for v1.6 which doesn't depend at runtime on `LibOSXUnwind_jll`, fix this in the registry so `libjulia_jll` never tries to pull in `LibOSXUnwind_jll`.  That'd solve all of our problems without hacks like this.

Should we keep this hack (I hope we don't), at very least please add a comment, I'm sure someone will be very confused by reading these lines without context in a few months.

Reply:
That seems like the way to go, but how is this dependency triggered, exactly? I don't see a `LibOSXUnwind_jll` in the `Project.toml` of `libjulia_jll v1.7.0+8`? Is it about removing these:

https://github.com/JuliaBinaryWrappers/libjulia_jll.jl/blob/2e64b99b0e204bf25308727beb971269affd3c4b/src/wrappers/x86_64-apple-darwin-julia_version%2B1.6.3.jl#L6

Reply:
It's only in v1.6

Reply:
Yes, but what I mean is that `libjulia_jll v1.7.0+8` is the one that [is getting installed](https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=16556&view=logs&j=11cc9737-c3b2-5b2e-49c1-3ecd4dd3473d&t=6864626c-b206-5daa-5d51-831805962223&l=169) (even when targeting Julia 1.6) and its `Project.toml` has no reference to `LibOSXUnwind_jll`: https://github.com/JuliaBinaryWrappers/libjulia_jll.jl/blob/main/Project.toml

Reply:
I am pretty sure `julia.h` from the libjulia_jll 1.6 won't be useable without `LibOSXUnwind_jll` because of the missing `libunwind.h` (either a compile error, or worse, it picks up a different "wrong" copy of `libunwind.h` somewhere). That said, we

Reply:
That said, we could perhaps indeed work around it, by either patching out the need for `libunwind.h` or by shipping a copy with it in the libjulia_jll -- after all, we only want this JLL as a build dependency

Looking at it, I think there is a realistic chance of doing even the patching part. Indeed, this patch might do it (applied in libjulai_jll *after* building it, i.e., either just before installing or even to the already install header files:
```patch
diff --git a/src/julia_internal.h b/src/julia_internal.h
index 0b76696791..092a7ca8dd 100644
--- a/src/julia_internal.h
+++ b/src/julia_internal.h
@@ -893,6 +893,7 @@ typedef struct {
 
 // Might be called from unmanaged thread
 uint64_t jl_getUnwindInfo(uint64_t dwBase);
+#if 0 // HACK to avoid libunwind dependency for libjulia_jll
 #ifdef _OS_WINDOWS_
 #include <dbghelp.h>
 JL_DLLEXPORT EXCEPTION_DISPOSITION __julia_personality(
@@ -933,8 +934,9 @@ size_t rec_backtrace_ctx(jl_bt_element_t *bt_data, size_t maxsize, bt_context_t
 #ifdef LIBOSXUNWIND
 size_t rec_backtrace_ctx_dwarf(jl_bt_element_t *bt_data, size_t maxsize, bt_context_t *ctx, jl_gcframe_t *pgcstack) JL_NOTSAFEPOINT;
 #endif
+#endif  // HACK to avoid libunwind dependency for libjulia_jll
 JL_DLLEXPORT jl_value_t *jl_get_backtrace(void);
-void jl_critical_error(int sig, bt_context_t *context);
+//void jl_critical_error(int sig, bt_context_t *context); // HACK to avoid libunwind dependency for libjulia_jll
 JL_DLLEXPORT void jl_raise_debugger(void);
 int jl_getFunctionInfo(jl_frame_t **frames, uintptr_t pointer, int skipC, int noInline) JL_NOTSAFEPOINT;
 JL_DLLEXPORT void jl_gdblookup(void* ip) JL_NOTSAFEPOINT;
diff --git a/src/julia_threads.h b/src/julia_threads.h
index f42984130f..6a30aa9008 100644
--- a/src/julia_threads.h
+++ b/src/julia_threads.h
@@ -67,7 +67,7 @@ typedef struct {
 #endif
 #if defined(JL_HAVE_UCONTEXT) || defined(JL_HAVE_UNW_CONTEXT)
 #define UNW_LOCAL_ONLY
-#include <libunwind.h>
+//#include <libunwind.h> // HACK to avoid libunwind dependency for libjulia_jll
 typedef struct {
     ucontext_t ctx;
 #if defined(JL_TSAN_ENABLED)
```

The idea here being that the disabled code really is internal and not relevant to any external consumer of the Julia kernel ABI. (It would be good if the Julia headers were restructured with this principle in mind anyway... *sigh*)

Reply:
What I don't understand is why it is looking for `LibOSXUnwind_jll`, when it is not even listed as a dependency in the `libjulia_jll` Project.toml. And from what I understand, this `delete!` call only removes it from the list of available standard libraries, not from the dependencies?

Reply:
@fingolfin I did some extra tests, and neither `libcxxwrap-julia` nor `jlqml` (compiled using the old binarybuilder) link to `libOSXUnwind.dylib`. I also tested the `libjlqml.dylib` built using the `delete!` hack, and it passes the QML tests. So I think this patch is not really needed, there is already no real dependency on `LibOSXUnwind_jll` and it is not used anyway (by libcxxwrap-julia and its dependents at least).

So I think it is safe to leave this hack in for now, to at least move forward with Julia 1.7-compatible binaries. I'll also update https://github.com/JuliaLang/Pkg.jl/issues/2942 with this info, since I have no idea what exactly is going on in Pkg here or why `LibOSXUnwind_jll` is even pulled into the dependency list.

Reply:
OK, so I guess the possibility of corrupt binaries is excluded, at worst we should get a linking error. I have added a comment, I'd propose to merge this and then investigate https://github.com/JuliaLang/Pkg.jl/issues/2942 further, libjulia_jll also doesn't build because of that right now.

Reply:
I agree, let's test this, if it works, this is a good short term solution that we can use for the time being also elsewhere.

Reply:
OK, from https://github.com/JuliaLang/Pkg.jl/pull/2963 I also conclude that this hack will in this case actually have the same effect as the (possible) fix. `LibOSXUnwind_jll` is here only encountered in the historical dependencies and then triggers a bug in the dependency resolution. So, @giordano this is now good to merge I think :)

Reply:
Ok, let's do it